apps/gyrokinetic.c

```diff
@@ -2212,29 +1173,15 @@ gkyl_gyrokinetic_app_write_species_conf(gkyl_gyrokinetic_app* app, int sidx, dou
 
   gkyl_gyrokinetic_app_write_species_rad_emissivity(app, sidx, tm, frame);
 
-  struct gk_species *gk_s = &app->species[sidx];
-  for (int j=0; j<gk_s->react.num_react; ++j) {
-    if ((gk_s->react.react_id[j] == GKYL_REACT_IZ) 
-      && (gk_s->react.type_self[j] == GKYL_SELF_ELC)) {
-      gkyl_gyrokinetic_app_write_species_iz_react(app, sidx, j, tm, frame);
-    }
-    if ((gk_s->react.react_id[j] == GKYL_REACT_RECOMB) 
-      && (gk_s->react.type_self[j] == GKYL_SELF_ELC)) {
-      gkyl_gyrokinetic_app_write_species_recomb_react(app, sidx, j, tm, frame);
+  struct gk_species *gks = &app->species[sidx];
+  for (int j=0; j<gks->react.num_react; ++j) {
+    if (gks->react.type_self[j] == GKYL_SELF_ION) {
+      gkyl_gyrokinetic_app_write_species_react(app, sidx, j, tm, frame);
     }
   }
-  for (int j=0; j<gk_s->react_neut.num_react; ++j) {
-    if ((gk_s->react_neut.react_id[j] == GKYL_REACT_IZ) 
-      && (gk_s->react_neut.type_self[j] == GKYL_SELF_ELC)) {
-      gkyl_gyrokinetic_app_write_species_iz_react_neut(app, sidx, j, tm, frame);
-    }
-    if ((gk_s->react_neut.react_id[j] == GKYL_REACT_RECOMB) 
-      && (gk_s->react_neut.type_self[j] == GKYL_SELF_ELC)) {
-      gkyl_gyrokinetic_app_write_species_recomb_react_neut(app, sidx, j, tm, frame);
-    }
-    if ((gk_s->react_neut.react_id[j] == GKYL_REACT_CX)
-      && (gk_s->react_neut.type_self[j] == GKYL_SELF_ION)) {
-      gkyl_gyrokinetic_app_write_species_cx_react_neut(app, sidx, j, tm, frame);
+  for (int j=0; j<gks->react_neut.num_react; ++j) {
+    if (gks->react_neut.type_self[j] == GKYL_SELF_ION) {
+      gkyl_gyrokinetic_app_write_species_react_neut(app, sidx, j, tm, frame);
```

Comment:
@tnbernard can you push this if-statement into write_species_react_neut? because from the gyrokinetic.c point of view, nearly all the write functions are very clean and assume no knowledge of the module they come from, except for this one; the user of `gkyl_gyrokinetic_app_write_species_react_neut` needs to know something about `react_neut.type_self` so it'd be good to push this down into the react module

Reply:
same thing in `gkyl_gyrokinetic_app_write_neut_species_source_mom` below